map_model/src/road.rs

```diff
@@ -113,11 +108,30 @@ pub struct Road {
     pub dst_i: IntersectionID,
 }
 
+type HomogenousTuple2<T> = (T, T);
+
 impl Road {
-    pub fn get_lane_types(&self) -> (Vec<LaneType>, Vec<LaneType>) {
+    pub fn children(&self, fwds: bool) -> &Vec<(LaneID, LaneType)> {
+        if fwds {
+            &self.children_forwards
+        } else {
+            &self.children_backwards
+        }
+    }
+
+    pub fn children_mut(&mut self, fwds: bool) -> &mut Vec<(LaneID, LaneType)> {
+        if fwds {
+            &mut self.children_forwards
+        } else {
+            &mut self.children_backwards
+        }
+    }
+
+    pub fn get_lane_types<'slf>(&'slf self) -> HomogenousTuple2<impl Iterator<Item=LaneType> + 'slf> {
+        let get_lanetype = |(_, lanetype): &(_, LaneType)| *lanetype;
```

Comment:
I'm not sure how I feel about this change. It adds a layer of abstraction to understand what's going on, and I don't know if it provides much value. What's the motivation here?

Reply:
Again, we can avoid `collect`ing into intermediate data structures where it is not strictly needed.